sympy/core/power.py

```diff
@@ -647,7 +649,8 @@ def _eval_is_extended_real(self):
 
         if real_b is False:  # we already know it's not imag
             i = arg(self.base)*self.exp/S.Pi
-            return i.is_integer
+            if i.is_complex: # finite
```

Comment:
Why don't you say `i.is_finite` here?

Reply:
It still passes the arit tests.

Reply:
The question is what would happen if `i` was somehow:
```
Symbol('x', finite=True, complex=False)
```
I don't really know what an object satisfying those assumptions would be though (a quaternion?)

Reply:
ok